src/setup.jl

```diff
@@ -212,13 +212,17 @@ function build_JuliaInterface(sysinfo::Dict{String, String})
     # paths involve spaces, but then we likely will haves problem in other
     # places; in any case, if anybody ever cares about this, we can work on
     # finding a better solution.
-    JULIA_CPPFLAGS = filter(c -> c != '\'', cflags())
+    JULIA_CFLAGS = filter(c -> c != '\'', cflags())
     JULIA_LDFLAGS = filter(c -> c != '\'', ldflags())
     JULIA_LIBS = filter(c -> c != '\'', ldlibs())
+    if get(ENV, "JULIAINTERFACE_WITH_COVERAGE", "false") == "true"
+        JULIA_CFLAGS *= " --coverage"
+        JULIA_LDFLAGS *= " --coverage"
+    end
```

Comment:
Instead of this limited special case, I'd prefer if we changed this code to not overwrite existing CFLAGS/LDFLAGS in the `withenv` below but rather to append to them.

Reply:
That was a 50/50 coin flip which of the two versions you prefered. This is in a stash and I can push it tomorrow